include/arch/xtensa/xtensa_api.h

```diff
@@ -7,7 +7,9 @@
 #define ZEPHYR_ARCH_XTENSA_INCLUDE_XTENSA_API_H_
 
 #include <xtensa/hal.h>
+#if defined(CONFIG_SOC_ESP32)
 #include <xtensa/core-macros.h>
+#endif
```

Comment:
I would actually say that the xtensa/core-macros.h header is wrong here. The commit, that added it didn't otherwise modify xtensa_api.h. So, either core-macros.h was needed there before and that commit was fixing a bug, or some other file, that includes xtensa_api.h from that commit, now after changes also needs core-macros.h. The former case is impossible, since now if we add that header on Intel ADSP platform build breaks. In the latter case it's those files, that have to include the header explicitly. It seems to me that it's drivers/entropy/entropy_esp32.c that needs it for `XTHAL_GET_CCOUNT()`. I think the core-macros.h header should be removed here and moved to that file.

Reply:
Yes, correct. `XTHAL_GET_CCOUNT()`is needed. I am checking as well.

Reply:
fine with me, this was just a workaround anyways to get things building again.